src/main/java/data/GCSDatabase.java

```diff
@@ -0,0 +1,165 @@
+  private Storage storage;
+
+  /* ProjectID. */
+  private String projectId = "pose-3d-viewer-step-2020";
+
+  /* Used below to determine the size of chucks to read in. Should be > 1kb and < 10MB. */
+  private static final int BUFFER_SIZE = 2 * 1024 * 1024;
+
+  /* These two next variables are copied from GCS tutorials. */
+  /*
+   * See the StorageClass documentation for other valid storage classes:
+   * https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/storage/StorageClass.html
+   */
+  private StorageClass storageClass = StorageClass.STANDARD;
+
+  /**
+   * This is where backoff parameters are configured. Here it is aggressively retrying with backoff,
+   * up to 10 times but taking no more that 15 seconds total to do so.
+   */
+  private final GcsService gcsService =
+      GcsServiceFactory.createGcsService(
+          new RetryParams.Builder()
+              .initialRetryDelayMillis(10)
+              .retryMaxAttempts(10)
+              .totalRetryPeriodMillis(15000)
+              .build());
+
+  //  private Credentials credentials;
+
+  /*
+   * See this documentation for other valid locations:
+   * http://g.co/cloud/storage/docs/bucket-locations#location-mr
+   */
+  private String location = "US";
+
+  /* Initiates GCSDatabase. */
+  public GCSDatabase(String name) {
+    System.out.println("in constructor");
+
+    System.out.println("initializing storage");
+
+    storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    System.out.println("making bucket");
+    bucketName = name;
+    bucket = storage.get(bucketName);
+
+    if (bucket == null) bucket = storage.create(BucketInfo.of(bucketName));
+    System.out.println("Created bucket " + bucket.getName());
+  }
+
+  /* Returns database name.*/
+  @Override
+  public String getDatabaseName() {
+    return bucketName;
+  }
+
+  /* Returns the appropiate and unique name for a particular file. */
+  private String name(String runId, String type) {
+    return runId + "_" + type;
+  }
+
+  /* Writes file into GCSDatabase. */
+  @Override
+  public OutputStream writeData(String runId, String type) throws IOException {
+    String objectName = name(runId, type);
+    return uploadFile(objectName);
+  }
+
+  /* uploads file with objectname to GCSDatabase. */
+  /*private OutputStream uploadFile(String objectName) throws IOException {
+    GcsFilename filename = new GcsFilename(bucketName, objectName);
+    GcsFileOptions instance = GcsFileOptions.getDefaultInstance();
+    GcsOutputChannel outputChannel;
+    outputChannel = gcsService.createOrReplace(filename, instance);
+    return Channels.newOutputStream(outputChannel);
+  }*/
+  private OutputStream uploadFile(String objectName) throws IOException {
+
+    Blob blob = bucket.create(objectName, new byte[0]);
+    WriteChannel writeChannel = blob.writer();
+    OutputStream out = Channels.newOutputStream(writeChannel);
+    System.out.println(out);
+    blob = blob.reload();
```

Comment:
I don't think you need `blob.reload()` since `blob` is a local variable. (Also, I think the examples suggest to call `reload` after writing to the bucket, but in your case the write doesn't happen within this function.)